docs/what-is-beergarden.adoc

```diff
@@ -1,64 +1,63 @@
 
-TIP: An example of a system could be something like a Calculator
+Commands::
+	The definition of an action, analogous to functions. The addition operation on a calculator.
 
-=== Commands
+Parameters::
+	Inputs to a command, analogous to function parameters. The numbers to be added together in the addition operation.
 
-If you are familiar with programming at all, commands are synonymous with methods. If you are not familiar with programming, commands are simply the definition of the action that you would like to take, not the action itself. Commands belong to a system. Commands may or may not have parameters.
+Requests::
+	An actual action that is performed, analogous to calling a function. The act of adding two numbers using the addition operation.
 
-TIP: An example of a command could be something like the '+' operator on a
-Calculator
 
-=== Parameters
+== An Example
 
-Parameters describe the type of information you can give to a command. This includes type, max/min values, regexes and lots more. Basically, parameters are the documentation that people want when they want to understand how to use the system/command you've created.
+So let's use those definitions to expand the calculator example a little. You want to give an end user the ability to add two numbers together and see the result.
 
-TIP: An example of a parameter could be something like 'x' when adding x + 1
+In Beer Garden you would have a System with name "Calculator". The Calculator System would have a single Command, called "add". The "add" command would accept two Parameters, let's call them "x" and "y". The "add" Command would implement the logic to turn the two Parameters into the correct result. In order to exercise the functionality provided by the "add" Command the user can create a Request with concrete values for the "x" and "y" Parameters. When that Request has been processed the result can be determined by looking at the Request's output.
 
-=== Requests
+Ok, that was admittedly a lot. Showing the code underlying the Calculator should make things clearer:
 
-Requests represent the actions that you actually want to take on the system. A request must detail what system it would like to take action on, what command to run, and what parameters to use on that command.
+[source,python]
+----
+class Calculator: <1>
 
-TIP: An example of a request would be like pressing '1', '+', '1', '=' on a
-calculator
+    def add(self, x, y): <2, 3>
+        return x + y
+----
+<1> Define the "Calculator" System
+<2> Define the "add" Command
+<3> Define the "x" and "y" Parameters
 
-=== Quick Example
+Now, if the end user wanted to add 1 and 1, they would issue a Request to Beer Garden. The most common way is to use the web UI that Beer Garden creates for this command. There will be input fields for "x" and "y", and a big "Make Request" button. Clicking that will take the user to a page where they'll be shown the result of their Request.
```

Comment:
It's probably worth showing the CURL request right here to exercise the calculator.

Reply:
So I added the curl command, but then I thought if we're adding the curl command, there's no reason to not add the SystemClient way as well. Especially since the curl command is not the cleanest - you need to specify the BG connection information as well.

And then I realized I had basically reproduced /docs/api-users-guide/#requests

And after thinking more, I think the actual curl command is too detailed for the "What is Beergarden?" page. However, I do think it's important to highlight that the UI is not the ONLY way to create a request here though, since that's a big part of "What is Beergarden?"

Also, the link to the Users Manual on the next line is broken, so it was good to check this.